### PR TITLE
working on serialization of World and Creature

### PR DESCRIPTION
**Pull request by gilesknap**
Because Creature contains `Box<dyn Genotype>`, I'm trying to use erased-serde to do the serialization.

But I'm struggling to see how to derive from erased-serde::Serialize.

### CODE REVIEW

**Comment by gilesknap** [comment]
@GDYendell @garryod any ideas?

See https://github.com/gilesknap/eyes2/blob/6a1f8e736af66b2b1086efd71cbfa38d03adb6ed/eyes2-lib/src/entity/creature.rs#L38-L46

**Comment by garryod** [comment]
Looks like you've missed the `#[macro_use] extern crate erased_serde;` and the `serialize_trait_object!` macro on `Genotype`. Not sure about serializing a `mpsc::Sender` though

**Comment by gilesknap** [comment]
Thanks @garryod got this working. But managed to confuse myself quite a bit in getting there.

File: eyes2-lib/src/entity/genotype/genotype.rs [hunk 1]
```diff
@@ -13,7 +11,7 @@ pub enum BadGenomeError {
 // expected that the Genotype will be defined by a genome, and that the
 // genome (with mutations as appropriate) will be passed to the
 // descendant creatures.
-pub trait Genotype {
+pub trait Genotype: erased_serde::Serialize {
```
Comment (garryod) [inline eyes2-lib/src/entity/genotype/genotype.rs:14]: This trait bound shouldn't be required afaik

Comment (gilesknap) [reply eyes2-lib/src/entity/genotype/genotype.rs:14]: apparently `erased_serde::serialize_trait_object!(Genotype)` requires it.

**Pull request merged**